From 4c8d86fa18a25e46d363a676b4c8c81404ffb933 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 12 Dec 2024 16:07:57 +0800
Subject: [PATCH 307/316] AOSCOS: init: make NUMA_BALANCING depend on
 TRANSPARENT_HUGEPAGE if MIPS

This should suppress an assertion failure during build. Further
investigation should be conducted when possible.

Fixes: "BACKPORT: FROMLIST: mips/mm: Add NUMA balancing support"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 init/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/init/Kconfig b/init/Kconfig
index a20e6efd3f0f..3ab90847416c 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -937,6 +937,7 @@ config NUMA_BALANCING
 	depends on ARCH_SUPPORTS_NUMA_BALANCING
 	depends on !ARCH_WANT_NUMA_VARIABLE_LOCALITY
 	depends on SMP && NUMA && MIGRATION && !PREEMPT_RT
+	depends on TRANSPARENT_HUGEPAGE || !MIPS
 	help
 	  This option adds support for automatic NUMA aware memory/task placement.
 	  The mechanism is quite primitive and is based on migrating memory when
-- 
2.48.1

